static/app/components/button.tsx

```diff
@@ -165,9 +189,43 @@ function BaseButton({
         return;
       }
 
+      const iconType = icon?.type.displayName;
+      const buttonLabel =
```

Comment:
don't we already extract the label somewhere higher up?

Reply:
Yeah @roggenkemper you should be able to just use `screenReaderLabel` for the `text` property. Buttons that only have an icon should already have an `aria-label` so I don't think there's any real need to check for the icon type.